tests/mockdata.json

```diff
@@ -1,24 +1,170 @@
+    },
+    "expected": {
+      "codename": "trusty",
+      "dist": "Ubuntu Linux",
+      "os": "linux",
+      "release": "14.04"
+    }
+  },
+  {
+    "desc": "Ubuntu 12.04",
+    "platform": "linux",
+    "file": {
+      "/etc/lsb-release": "DISTRIB_ID=Ubuntu\nDISTRIB_RELEASE=12.04\nDISTRIB_CODENAME=precise\nDISTRIB_DESCRIPTION=\"Ubuntu 12.04.5 LTS\"\n"
+    },
+    "expected": {
+      "codename": "precise",
+      "dist": "Ubuntu Linux",
+      "os": "linux",
+      "release": "12.04"
+    }
+  },
+  {
+    "desc": "Fedora 21",
+    "platform": "linux",
+    "file": {
+      "/etc/fedora-release": "Fedora release 21 (Twenty One)\n"
+    },
+    "expected": {
+      "codename": "Twenty One",
+      "dist": "Fedora",
+      "os": "linux",
+      "release": "21"
+    }
+  },
+  {
+    "desc": "CentOS 5.11",
+    "platform": "linux",
+    "file": {
+      "/etc/redhat-release": "CentOS release 5.11 (Final)\n"
+    },
+    "expected": {
+      "codename": "final",
+      "dist": "Centos",
+      "os": "linux",
+      "release": "5.11"
+    }
+  },
+  {
+    "desc": "Centos 6.6",
+    "platform": "linux",
+    "file": {
+      "/etc/redhat-release": "CentOS release 6.6 (Final)\n"
+    },
+    "expected": {
+      "codename": "final",
+      "dist": "Centos",
+      "os": "linux",
+      "release": "6.6"
+    }
+  },
+  {
+    "desc": "Centos 7.1",
+    "platform": "linux",
+    "file": {
+      "/etc/redhat-release": "CentOS Linux release 7.1.1503 (Core)\n"
+    },
+    "expected": {
+      "codename": "core",
+      "dist": "Centos",
+      "os": "linux",
+      "release": "7.1.1503"
+    }
+  },
+  {
+    "desc": "RHEL 6.5",
+    "platform": "linux",
+    "file": {
+      "/etc/redhat-release": "Red Hat Enterprise Linux Server release 6.5 (Santiago)\n"
+    },
+    "expected": {
+      "codename": "santiago",
+      "dist": "Red Hat Linux",
+      "os": "linux",
+      "release": "6.5"
+    }
+  },
+  {
+    "desc": "RHEL 6.6",
+    "platform": "linux",
+    "file": {
+      "/etc/redhat-release": "Red Hat Enterprise Linux Server release 6.6 (Santiago)\n"
+    },
+    "expected": {
+      "codename": "santiago",
+      "dist": "Red Hat Linux",
+      "os": "linux",
+      "release": "6.6"
+    }
+  },
+  {
+    "desc": "RHEL 7.0",
+    "platform": "linux",
+    "file": {
+      "/etc/redhat-release": "Red Hat Enterprise Linux Server release 7.0 (Maipo)\n"
+    },
+    "expected": {
+      "codename": "maipo",
+      "dist": "Red Hat Linux",
+      "os": "linux",
+      "release": "7.0"
+    }
+  },
+  {
+    "desc": "SUSE Linux",
+    "platform": "linux",
+    "file": {
+      "/etc/SuSE-release": "SUSE Linux Enterprise Server 11 (x86_64)\nVERSION = 11\nPATCHLEVEL = 4\n"
+    },
+    "expected": {
+      "dist": "SUSE Linux",
+      "os": "linux",
+      "release": "11"
+    }
+  },
+  {
+    "desc": "Alpine Linux",
+    "platform": "linux",
+    "file": {
+      "/etc/alpine-release": "3.3\n"
```

Comment:
Thanks for this! Don't worry about getting the Docker tests to pass, I'm contemplating getting rid of those since they take so long to run.

If you get this test to pass, I'll accept the PR :smile: